README.md

```diff
@@ -57,3 +57,27 @@ To get Camera Images,run:
 A python script, ```takePhoto.py``` was written as a subscriber node to save the image automatically and the publisher node included in the Gazebo model launch file.
 
 P.S: This work would continually be improved upon and the whole of the robot would move , detect obstacles and change direction of motion.
+
+### Unit Test
+
+Forward Kinematics was implemneted for 6 joints robotic system as a model in FKine.py.
+
+In FKtest.py, some teste cases were tested using the function from FKine and their actual expected values compared with the solution using numpy asserts.
+Several test cases can be added to implement thus unit test.
+To test this unit test implementation, simply clone thr repository and navigate to script directory where we have the forward kinematics implementation and its unit test. Thereafter from the terminal, run the script using
+
+
+```python FKtest.py```
+
+
+### Integration Test
+
+
+The integration tests is a continuation of the joint movement/control task.
+Firstly, from previous task, some joints poitions are published to robot joints. In other to get the actual joint position, an IMU sensor was added to the urdf file as a Gazebo plugin. Thereafter a subscriber was implemneted to determine if actual movement takes place which is determined by reading accelearation information from the robot joint. The integration test confirms if actual movement takes place when some joint positions are published. 
+
```

Comment:
hm, now i see, but better to read joint_states, because sensor can work and send values but the positions is still zero.

Reply:
yes, position I still zero. I actually didn't figure out how to read exact joint positions. So i decided to measure linear acceleration. So the test is basically checking if linear acceleration is no longer zero once joint commands have been sent.